plans/objchange/objchange.go

```diff
@@ -102,181 +279,176 @@ func proposedNewObject(schema *configschema.Block, prior, config cty.Value) cty.
-					for it := configV.ElementIterator(); it.Next(); {
-						idx, configEV := it.Element()
-						k := idx.AsString()
-						if priorV.IsKnown() && (priorV.IsNull() || !priorV.HasIndex(idx).True()) {
-							// If there is no corresponding prior element then
-							// we just take the config value as-is.
-							newVals[k] = configEV
-							continue
-						}
-						priorEV := priorV.Index(idx)
-
-						newEV := ProposedNewObject(&blockType.Block, priorEV, configEV)
-						newVals[k] = newEV
-					}
-					newV = cty.MapVal(newVals)
-				} else {
-					newV = cty.MapValEmpty(blockType.ImpliedType())
-				}
-			}
-
-		case configschema.NestingSet:
-			if !configV.Type().IsSetType() {
-				panic("configschema.NestingSet value is not a set as expected")
-			}
-
-			// Nested blocks are correlated by comparing the element values
-			// after eliminating all of the computed attributes. In practice,
-			// this means that any config change produces an entirely new
-			// nested object, and we only propagate prior computed values
-			// if the non-computed attribute values are identical.
-			var cmpVals [][2]cty.Value
-			if priorV.IsKnown() && !priorV.IsNull() {
-				cmpVals = setElementCompareValues(&blockType.Block, priorV, false)
+				newV = cty.EmptyObjectVal
 			}
+		} else {
 			configVLen := 0
-			if configV.IsKnown() && !configV.IsNull() {
-				configVLen = configV.LengthInt()
+			if config.IsKnown() && !config.IsNull() {
+				configVLen = config.LengthInt()
 			}
 			if configVLen > 0 {
-				used := make([]bool, len(cmpVals)) // track used elements in case multiple have the same compare value
-				newVals := make([]cty.Value, 0, configVLen)
-				for it := configV.ElementIterator(); it.Next(); {
-					_, configEV := it.Element()
-					var priorEV cty.Value
-					for i, cmp := range cmpVals {
-						if used[i] {
-							continue
-						}
-						if cmp[1].RawEquals(configEV) {
-							priorEV = cmp[0]
-							used[i] = true // we can't use this value on a future iteration
-							break
-						}
-					}
-					if priorEV == cty.NilVal {
-						priorEV = cty.NullVal(blockType.ImpliedType())
+				newVals := make(map[string]cty.Value, configVLen)
+				for it := config.ElementIterator(); it.Next(); {
+					idx, configEV := it.Element()
+					k := idx.AsString()
+					if prior.IsKnown() && (prior.IsNull() || !prior.HasIndex(idx).True()) {
+						// If there is no corresponding prior element then
+						// we just take the config value as-is.
+						newVals[k] = configEV
+						continue
 					}
+					priorEV := prior.Index(idx)
 
-					newEV := ProposedNewObject(&blockType.Block, priorEV, configEV)
-					newVals = append(newVals, newEV)
+					newEV := proposedNewAttributes(schema.Attributes, priorEV, configEV)
+					newVals[k] = cty.ObjectVal(newEV)
 				}
-				newV = cty.SetVal(newVals)
+				newV = cty.MapVal(newVals)
 			} else {
-				newV = cty.SetValEmpty(blockType.Block.ImpliedType())
+				newV = cty.MapValEmpty(schema.ImpliedType())
```

Comment:
And again here, I think we want a case for a null value rather than an empty value. Just like the tuple case above, I'm not sure what the object case will be, but I'll see what the tests look like.